### PR TITLE
feat(run-task-handler): added hook in finally block that will be ran regardless of whether the task completed without exception.

### PR DESCRIPTION
**Pull request by fieldju**
feat(run-task-handler): added hook in finally block that will be run regardless of whether the task completed without exception.

The existing hook doesn't execute when a task execution throws an exception.
This hook allows for an extension project to clear the security context that is set in the before task hook.

### CODE REVIEW

File: orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/RunTaskHandler.kt [hunk 1]
```diff
@@ -189,13 +190,15 @@ class RunTaskHandler(
                   log.error("Error running ${message.taskType.simpleName} for ${message.executionType}[${message.executionId}]", e)
                 }
               }
-
+              taskException = e;
```
Comment (cfieber) [inline orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/RunTaskHandler.kt:193]: is there a reason not to set taskException for the other cases in the catch block ?

Comment (fieldju) [reply orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/RunTaskHandler.kt:193]: My original thought was that those other blocks retry, but thinking about it more no, it should be set right away.

I'll move it up.

File: orca-core/src/main/java/com/netflix/spinnaker/orca/TaskExecutionInterceptor.java [hunk 2]
```diff
@@ -50,4 +50,7 @@ default StageExecution beforeTaskExecution(Task task, StageExecution stage) {
   default TaskResult afterTaskExecution(Task task, StageExecution stage, TaskResult taskResult) {
     return taskResult;
   }
+
+  default void finallyAfterTaskExecution(
```
Comment (cfieber) [inline orca-core/src/main/java/com/netflix/spinnaker/orca/TaskExecutionInterceptor.java:54]: can you add a doc comment here on the purpose of this hook point for implementers

**Review by cfieber — APPROVED** [review]
LGTM would just like to see javadoc on the new interface method

File: orca-core/src/main/java/com/netflix/spinnaker/orca/TaskExecutionInterceptor.java (hunk 2 above)
Comment (fieldju) [reply orca-core/src/main/java/com/netflix/spinnaker/orca/TaskExecutionInterceptor.java:66]: - [X] add docs

**Pull request merged**